### PR TITLE
pre-commit: check more file types for trailing whitespace

### PR DESCRIPTION
**Pull request by jbampton**
### Description

This PR checks more file types for unneeded trailing whitespace.

Trailing whitespace is just extra data that we don't need.

This hook when run locally will automatically trim the trailing whitespace.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [X] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by jbampton** [comment]
@blueorangutan package

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -59,8 +59,9 @@ repos:
       - id: mixed-line-ending
         exclude: \.(cs|xml)$
       - id: trailing-whitespace
-        files: \.(in|java|md|py|rb|sh|vue|yaml|yml)$
+        files: \.(header|in|java|md|properties|py|rb|sh|sql|txt|vue|yaml|yml)$
```
Comment (DaanHoogland) [inline .pre-commit-config.yaml:62]: any .header files?

**Review by DaanHoogland — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jbampton) [reply .pre-commit-config.yaml:62]: At least one 

https://github.com/apache/cloudstack/blob/main/LICENSE.header

Comment (DaanHoogland) [reply .pre-commit-config.yaml:62]: ah, ok, it wasn't mentioned in this PR, hence my question

**Review by JoaoJandre — APPROVED** [review]

**Pull request merged**